lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -202,6 +203,25 @@ def _check_state():
 
         if _check_state():
             proc = module.run_command([module.get_bin_path('ssh-keygen', True), '-lf', self.path])
+            if not proc[0] == 0:
+                if not os.path.isdir(self.path) and os.access(self.path, os.W_OK):
+                    if not self.force:
+                        module.fail_json(
+                            msg='%s seems not to be a valid key file. If you want this module to overwrite '
+                            'the file at %s please specify the force parameter.' % (self.path, self.path))
+                    else:
+                        return False
+                elif not os.path.isdir(self.path) and not os.access(self.path, os.W_OK):
+                    if not self.force:
+                        module.fail_json(
+                            msg='%s seems not to be write-able. If you want this module to (re)generate the '
```

Comment:
The module should not wait for `force` here. All `openssl_*` modules also don't do that, including `openssl_privatekey`.